Main/Source/gods.cpp

```diff
@@ -126,15 +129,166 @@ int mortifer::GetBasicAlignment() const { return EVIL; }
+  FAVOUR_CUREPOISON,
+  FAVOUR_CURESLOWNESS,
+  FAVOUR_CURETAPEWORM,
+  FAVOUR_CUREWOUNDS,
+  FAVOUR_DISEASEIMMUNITY,
+  FAVOUR_EARTHQUAKE,
+  FAVOUR_ENCHANT,
+  FAVOUR_ETHEREALMOV,
+  FAVOUR_EXTINGUISHFIRE,
+  FAVOUR_FEED,
+  FAVOUR_FIRESTORM,
+  FAVOUR_FIXEQUIPMENT,
+  FAVOUR_HEALBURNS,
+  FAVOUR_HOLYGREN,
+  FAVOUR_INFRAVISION,
+  FAVOUR_INVIGORATE,
+  FAVOUR_INVISIBILITY,
+  FAVOUR_SHOPPING,
+  FAVOUR_SPEEDUP,
+  FAVOUR_STOPFIRE,
+  FAVOUR_SUMMONWOLF,
+  FAVOUR_TAME,
+  FAVOUR_TELEPORT,
+  FAVOUR_BURNENEMIES,
+  FAVOUR_FEELENEMIES,
+  FAVOUR_POLYCONTROL,
+  FAVOUR_TELEPCONTROL,
+  FAVOUR_ENRAGE,
+  FAVOUR_CAUSEFEAR,
+};
+
+void god::FavourInit() //this one is better on this file
+{
+  AddFavourID(FAVOUR_BURNENEMIES,"Burn your Enemies");
+  AddFavourID(FAVOUR_CALLRAIN,"Make it Rain");
+  AddFavourID(FAVOUR_CAUSEFEAR,"Your enemies will Fear you");
+  AddFavourID(FAVOUR_CONFUSE,"Cause Confusion amongst your enemies");
+  AddFavourID(FAVOUR_CURELEPROSY,"Cure Leprosy");
+  AddFavourID(FAVOUR_CURELYCANTHROPY,"Cure Lycanthropy");
+  AddFavourID(FAVOUR_CUREMINDWORM,"Cure Mindworm");
+  AddFavourID(FAVOUR_CUREPOISON,"Cure Poison");
+  AddFavourID(FAVOUR_CURESLOWNESS,"Cure Slowness");
+  AddFavourID(FAVOUR_CURETAPEWORM,"Cure Tapeworm");
+  AddFavourID(FAVOUR_CUREWOUNDS,"Cure Wounds");
+  AddFavourID(FAVOUR_DISEASEIMMUNITY,"Gain temporary Immunity to Diseases");
+  AddFavourID(FAVOUR_EARTHQUAKE,"Invoke the rage of an Earth Quake");
+  AddFavourID(FAVOUR_ENCHANT,"Enchant Equipment");
+  AddFavourID(FAVOUR_ENRAGE,"Fill you with Rage");
+  AddFavourID(FAVOUR_ETHEREALMOV,"Become Ethereal");
+  AddFavourID(FAVOUR_EXTINGUISHFIRE,"Put out these Flames"); //TODO consider price vs FAVOUR_HEALBURNS);
+  AddFavourID(FAVOUR_FEED,"Calms your Hunger");
+  AddFavourID(FAVOUR_FEELENEMIES,"Feel your Enemies");
+  AddFavourID(FAVOUR_FIRESTORM,"Fiery Firestorm");
+  AddFavourID(FAVOUR_FIXEQUIPMENT,"Fix one broken equipped item");
+  AddFavourID(FAVOUR_HEALBURNS,"Heals your burns");
+  AddFavourID(FAVOUR_HOLYGREN,"Paladin's Holy Grenade");
+  AddFavourID(FAVOUR_INFRAVISION,"See in the Darkness");
+  AddFavourID(FAVOUR_INVIGORATE,"Invigorate");
+  AddFavourID(FAVOUR_INVISIBILITY,"Become Invisible");
+  AddFavourID(FAVOUR_POLYCONTROL,"Control what you are");
+  AddFavourID(FAVOUR_SHOPPING,"Black Friday");
+  AddFavourID(FAVOUR_SPEEDUP,"Make you Fast");
+  AddFavourID(FAVOUR_STOPFIRE,"Unburn one Equipment");
+  AddFavourID(FAVOUR_SUMMONWOLF,"Summon Wolf friend(s)");
+  AddFavourID(FAVOUR_TAME,"Tame this Monster");
+  AddFavourID(FAVOUR_TELEPCONTROL,"Decide where you are sent");
+  AddFavourID(FAVOUR_TELEPORT,"Teleport");
+}
+
+int CalcDebit(god* G,int iDebit,int iDefault){
+  if(iDebit!=0){
+    switch(iDebit){
+      case FAVOURDEBIT_AUTO:       iDebit=iDefault  ;break;
+      case FAVOURDEBIT_AUTOHALF:   iDebit=iDefault/2;break;
+      case FAVOURDEBIT_AUTODOUBLE: iDebit=iDefault*2;break;
+    }
+    
+    if(G->GetRelation() < 0) //costy if bad relation
```

Comment:
I don't think you should be able to ask any favour of gods with whom you have negative relation. They should punish you for your insolence, rather.